ext/reflection/php_reflection.c

```diff
@@ -3114,6 +3114,8 @@ ZEND_METHOD(ReflectionMethod, getClosure)
 		} else {
 			zend_create_fake_closure(return_value, mptr, mptr->common.scope, Z_OBJCE_P(obj), obj);
 		}
+	} else {
+		zend_argument_value_error(1, "cannot be null for non-static methods");
```

Comment:
Structurally, I'd make the code above an `else` and add a `if (!obj)` check at the start.

Reply:
Done :)